wordpress-mobile/WordPress-iOS
# Reader post details: mark post as seen when viewed

Ref #15355 

This does two things:
1. When a post is viewed from a reader stream, it is automatically marked `seen`.
2. When a post's seen status changes from either of these locations, the post is updated in the stream so the card's menu has the correct seen option.
  - From viewing the post (point 1 above).
  - From manually toggling the seen status from post detail's options menu.

I also centralized the reader's NSNotification definitions into `ReaderHelpers`, hence the seemingly unrelated changes to the `ReaderTopicUnfollowed` notification.

To test:

---
Automatically marking posts as `seen`:
- In the Reader, tap on a post.
- Tap on the menu button in the nav bar.
  - Verify the seen option is `Mark as unseen` (since the post was automatically marked as `seen`).
- Don't select anything. Close the post details & return to the reader stream.
- Tap the menu button on the card.
  - Verify the seen option is `Mark as unseen`.

| Post details | Stream after post viewed  |
|--------|-------|
| ![post_details](https://user-images.githubusercontent.com/1816888/105247908-c49c4480-5b32-11eb-997d-27e84c079ad5.jpg) | ![stream](https://user-images.githubusercontent.com/1816888/105247917-c82fcb80-5b32-11eb-8d83-5905e34f0d03.jpg) |

---
Manually toggling seen:
- In the Reader, tap on a post.
- Tap on the menu button in the nav bar.
- Select `Mark as unseen`.
- Close the post details & return to the reader stream.
- Tap the menu button on the card.
  - Verify the seen option is `Mark as seen`.

Feel free to toggle seen in post details to whatever you like. The point here is that the card's menu reflects the final state in post details.

| Post details | Stream after toggling |
|--------|-------|
| ![post_details2](https://user-images.githubusercontent.com/1816888/105248316-6a4fb380-5b33-11eb-9d34-228fbd60f676.jpg) | ![stream2](https://user-images.githubusercontent.com/1816888/105248327-70459480-5b33-11eb-82f2-dab6771dd51d.jpg) |

---
PR submission checklist:
